### ISSUE TITLE
Could you please provide the details of val_data form ( bcolz.carray and list.npy) ?

### ISSUE THREAD

**Issue by vtddggg** [ZhaoJ9014/face.evoLVe.PyTorch#74]
👍 Great work !! Thanks for you repo, which helps me a lot !
In your code, the val_data is saved as `bcolz.carray` (images) and `numpy` (labels) form
I see you load them by:
`carray = bcolz.carray(rootdir = os.path.join(path, name), mode = 'r')`
`issame = np.load('{}/{}_list.npy'.format(path, name))`

and then extract features by:
```
emb_batch = backbone(ccropped.to(device)).cpu() + backbone(fliped.to(device)).cpu()
embeddings[idx:idx + batch_size] = l2_norm(emb_batch)
```

But you used part of embeddings to construct embedding1 and embedding2:
```
embeddings1 = embeddings[0::2]
embeddings2 = embeddings[1::2]
```

Not like that:
```
embeddings1 = embeddings
embeddings2 = embeddings
```

So, I have some doubt about why you construct embeddings like that way and what saved in `bcolz.carray`.
Also I will be appreciated if you can give an example of how to generate the `bcolz.carray` using raw data, such as raw LFW dataset.

**Comment by ZhaoJ9014** [comment]
> 👍 Great work !! Thanks for you repo, which helps me a lot !
> In your code, the val_data is saved as `bcolz.carray` (images) and `numpy` (labels) form
> I see you load them by:
> `carray = bcolz.carray(rootdir = os.path.join(path, name), mode = 'r')`
> `issame = np.load('{}/{}_list.npy'.format(path, name))`
> 
> and then extract features by:
> 
> ```
> emb_batch = backbone(ccropped.to(device)).cpu() + backbone(fliped.to(device)).cpu()
> embeddings[idx:idx + batch_size] = l2_norm(emb_batch)
> ```
> 
> But you used part of embeddings to construct embedding1 and embedding2:
> 
> ```
> embeddings1 = embeddings[0::2]
> embeddings2 = embeddings[1::2]
> ```
> 
> Not like that:
> 
> ```
> embeddings1 = embeddings
> embeddings2 = embeddings
> ```
> 
> So, I have some doubt about why you construct embeddings like that way and what saved in `bcolz.carray`.
> Also I will be appreciated if you can give an example of how to generate the `bcolz.carray` using raw data, such as raw LFW dataset.

Hi, the validation code is based on the evaluation protocol of the target dataset. Actually, you can organize your testing data similar to our training data (see Line 61-83 of train.py). For your question, you can refer to https://forums.fast.ai/t/how-to-use-bcolz-carray-when-data-set-is-too-large-to-load-into-memory/34813
